pkg/provider/resource_region.go

```diff
@@ -4,185 +4,243 @@ import (
+		Blocks: map[string]schema.Block{
+			"timeouts": timeouts.Block(ctx,
+				timeouts.Opts{Create: true, Delete: true, Update: true}),
 		},
 
-		Schema: map[string]*schema.Schema{
-			"cloud_provider": {
-				Description: "Cloud provider. For example, \"aws\" or \"azure\".",
-				Type:        schema.TypeString,
-				Required:    true,
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				MarkdownDescription: "Resource ID of the region.",
+				Computed:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
 			},
-			"project_id": {
-				Description:      "BigAnimal Project ID.",
-				Type:             schema.TypeString,
-				Required:         true,
-				ValidateDiagFunc: validateProjectId,
+			"cloud_provider": schema.StringAttribute{
+				MarkdownDescription: "Cloud provider. For example, \"aws\" or \"azure\".",
+				Required:            true,
 			},
-			"region_id": {
-				Description: "Region ID of the region. For example, \"germanywestcentral\" in the Azure cloud provider or \"eu-west-1\" in the AWS cloud provider.",
-				Type:        schema.TypeString,
-				Required:    true,
+			"project_id": schema.StringAttribute{
+				MarkdownDescription: "BigAnimal Project ID.",
+				Required:            true,
+				Validators: []validator.String{
+					ProjectIdValidator(),
+				},
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
 			},
-			"name": {
-				Description: "Region name of the region. For example, \"Germany West Central\" or \"EU West 1\".",
-				Type:        schema.TypeString,
-				Computed:    true,
+			"region_id": schema.StringAttribute{
+				MarkdownDescription: "Region ID of the region. For example, \"germanywestcentral\" in the Azure cloud provider or \"eu-west-1\" in the AWS cloud provider.",
+				Required:            true,
 			},
-			"status": {
-				Description: "Region status of the region. For example, \"ACTIVE\", \"INACTIVE\", or \"SUSPENDED\".",
-				Type:        schema.TypeString,
-				Optional:    true,
-				Default:     api.REGION_ACTIVE,
+			"name": schema.StringAttribute{
+				MarkdownDescription: "Region name of the region. For example, \"Germany West Central\" or \"EU West 1\".",
+				Computed:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
 			},
-			"continent": {
-				Description: "Continent that region belongs to. For example, \"Asia\", \"Australia\", or \"Europe\".",
-				Type:        schema.TypeString,
-				Computed:    true,
+			"status": schema.StringAttribute{
+				MarkdownDescription: "Region status of the region. For example, \"ACTIVE\", \"INACTIVE\", or \"SUSPENDED\".",
+				Optional:            true,
+				/*
```

Comment:
What will we do with this default part? @YanniHu1996

Reply:
sorry, this is wrong comment, I will recover